File: src/main/java/jisop/domain/Conventions.java
package jisop.domain;

import java.util.Arrays;
import java.util.HashSet;
import java.util.Set;
import java.util.TreeSet;

/**
 * Created by mathieu.
 */
public class Conventions {
    public static String ControllerName="controller";

    public static Set<String> ConfigurationName = IgnoreCase(new String[]{"isopconfiguration"});

    private static Set<String> IgnoreCase(String[] strings) {
        Set<String> s= new TreeSet<String>(String.CASE_INSENSITIVE_ORDER);
        s.addAll(Arrays.asList(strings));
        return s;
    }

    public static String Help="help";

    public static String Index ="index";

    public static boolean equalsHelp(String value){
        return Help.equals(value.toLowerCase());
    }
    public static boolean equalsIndex(String value){
        return Index.equals(value.toLowerCase());
    }
}

File: src/main/java/isop/domain/Conventions.kt
package isop.domain

import java.util.Arrays
import java.util.TreeSet

/**
 * Created by mathieu.
 */
object Conventions {
    var ControllerName = "controller"

    var ConfigurationName = IgnoreCase(arrayOf("isopconfiguration"))

    var Help = "help"

    var Index = "index"

    private fun IgnoreCase(strings: Array<String>): Set<String> {
        val s = TreeSet(String.CASE_INSENSITIVE_ORDER)
        s.addAll(Arrays.asList(*strings))
        return s
    }

    fun equalsHelp(value: String): Boolean {
        return Help == value.toLowerCase()
    }

    fun equalsIndex(value: String): Boolean {
        return Index == value.toLowerCase()
    }
}
